Update courier to push correctly formatted bundle

SamiSousa opened the pull request:
- Set data subfields to strings

This makes the pushed bundle correctly formatted, allowing the backend marketplace operator to properly reconcile the bundle.

/cc @kevinrizza

kevinrizza commented on operatorcourier/api.py:
```diff
@@ -60,6 +60,11 @@ def build_verify_and_push(namespace, repository, revision, token, source_dir=Non
+        # Set data fields to strings to match expected bundle format
+        bundle['data']['customResourceDefinitions'] = yaml.dump(bundle['data']['customResourceDefinitions'])
+        bundle['data']['clusterServiceVersions'] = yaml.dump(bundle['data']['clusterServiceVersions'])
+        bundle['data']['packages'] = yaml.dump(bundle['data']['packages']
```
Can you actually move this to the end of the build_and_verify() method so that method returns the correctly formatted bundle? I think it has to be after the validation so the fields can be referenced directly.

kevinrizza commented:
@SamiSousa Once you make the change I asked for the only other thing is the tests will fail so you will need to reformat the output bundles to make them pass (just six total characters per bundle I guess?)

SamiSousa commented:
Moved the format call to the end of `build_and_verify()`. Also updated the bundle test files to use strings for data fields rather than yaml lists. For the formatting to work I had to indent the content, so that accounts for most of the changes here.  
Since some tests load in bundles directly from files, I also implemented an unformat function, and moved all the formatting logic into format.py

SamiSousa commented on operatorcourier/format.py:
```diff
@@ -0,0 +1,86 @@
+def format_bundle(bundle):
+    """
+    Converts a bundle object into a push-ready bundle by changing list values of 'customResourceDefinitions', 'clusterServiceVersions', and 'packages' into stringified yaml literals.
+    This format is required by the Marketplace backend.
```
Yup, tested with current end to end flow on an operator

awgreene replied:
Wonderful.

SamiSousa commented on operatorcourier/format.py:
```diff
@@ -0,0 +1,86 @@
+            for csv in bundle['data']['clusterServiceVersions']:
+                if 'metadata' in csv:
+                    if 'annotations' in csv['metadata']:
+                        if 'alm-examples' in csv['metadata']['annotations']:
```
Formatting `alm-examples` and `description` isn't required for proper parsing by the marketplace operator, but I added it because (1) it's more readable in the resulting bundle and (2) I thought it would help with consistent format for the main two multiline string fields as part of the current test suite. Since it's not required I have no objections removing the check for these two fields, as long as tests can still pass.
I think it's worthwhile to create such a method that checks a path exists in a dict, would be a nice util function (if it doesn't exist in the standard lib).

awgreene replied:
Okay - I'll try and get around to adding the method in the future.
